### PR TITLE
vulnerabilities NSWG-ECO 450 to 452

### PR DESCRIPTION
**Pull request by dgonzalez**

### CODE REVIEW

File: vuln/npm/450.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "id": 387,
```
Comment (lirantal) [inline vuln/npm/450.json]: file is named 450 and id is 387

File: vuln/npm/450.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "title": "Priviledge escalation",
+  "author": "Patrick (https://twitter.com/patrickrbc)",
+  "module_name": "express-cart",
+  "publish_date": "2018-07-02",
+  "cves": [],
+  "vulnerable_versions": "<1.1.6",
+  "patched_versions": ">=1.1.7",
+  "slug": "express-cart-priviledge-escalation",
+  "overview": "A deficiency in the access control allows normal users from expressCart to add new users to the application including admin users.",
+  "recommendation": "update express-cart to 1.1.6 or higher",
```
Comment (lirantal) [inline vuln/npm/450.json]: should be 1.1.7 or higher

File: vuln/npm/451.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    "id": 388,
```
Comment (lirantal) [inline vuln/npm/451.json]: file name is different than id

File: vuln/npm/451.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "created_at": "2018-07-02",
+    "updated_at": "2018-07-02",
+    "title": "Stored XSS in the filename when directories listing",
+    "author": "tungpun (Twitter: https://twitter.com/tungpun_)",
+    "module_name": "glance",
+    "publish_date": "2018-07-02",
+    "cves": [],
+    "vulnerable_versions": "<=3.0.5",
+    "patched_versions": "",
```
Comment (lirantal) [inline vuln/npm/451.json]: we don't allow empty strings, use null

File: vuln/npm/451.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    "slug": "glance-filename-xss",
+    "overview": "A lack of output sanitization allows an attacker to run malicious javascript code in the user's browser",
+    "recommendation": "",
```
Comment (lirantal) [inline vuln/npm/451.json]: we don't allow recommendation

File: vuln/npm/452.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+    "id": 389,
```
Comment (lirantal) [inline vuln/npm/452.json]: file name is different than id

File: vuln/npm/452.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    "created_at": "2018-07-02",
+    "updated_at": "2018-07-02",
+    "title": "HTML Injection in filenames displayed as directory listing in the browser allows to embed iframe with malicious JavaScript",
+    "author": "bl4de (https://twitter.com/_bl4de)",
+    "module_name": "m-server",
+    "publish_date": "2018-07-02",
+    "cves": [],
+    "vulnerable_versions": "<1.4.2",
+    "patched_versions": ">=1.4.2",
+    "slug": "m-server-html-injection",
```
Comment (lirantal) [inline vuln/npm/452.json]: also we don't need the slugs

**Comment by lirantal** [comment]
@dgonzalez you may want to use the nswg-reporter script that I wrote to create those reports automatically.

**Comment by dgonzalez** [comment]
Yes, I had them stacked. Will do.

**Comment by lirantal** [comment]
@dgonzalez you might want to tidy it up before we're able to merge

**Pull request closed without merging**